src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/CalendarService.kt

```diff
@@ -19,16 +19,18 @@ class CalendarService(
   fun getCalendarInfo(user: UserEntity, premisesId: UUID, startDate: LocalDate, endDate: LocalDate): Map<CalendarBedInfo, List<CalendarOccupancyInfo>> {
     val calendarInfo = calendarRepository.getCalendarInfo(premisesId, startDate, endDate)
 
-    calendarInfo.entries.forEach { bed ->
-      bed.value.forEach {
-        if (it is CalendarBookingInfo) {
-          val offenderResult = offenderService.getOffenderByCrn(it.crn, user.deliusUsername, user.hasQualification(UserQualification.LAO))
+    val crns = calendarInfo.values.flatMap {
+        it.filterIsInstance<CalendarBookingInfo>().map { it.crn }
+    }
+    val offenderSummaries = offenderService.getOffenderSummariesByCrns(crns, user.deliusUsername, user.hasQualification(UserQualification.LAO), forceApDeliusContextApi = false)
 
-          it.personName = when (offenderResult) {
-            is AuthorisableActionResult.NotFound -> "Unknown"
-            is AuthorisableActionResult.Success -> "${offenderResult.entity.firstName} ${offenderResult.entity.surname}"
-            is AuthorisableActionResult.Unauthorised -> "LAO Offender"
-          }
+    calendarInfo.values.forEach { bedList ->
+      bedList.filterIsInstance<CalendarBookingInfo>().forEach { bed ->
+        val offenderSummary = offenderSummaries.first { it.crn == bed.crn }
+        bed.personName = when (offenderSummary) {
+          is PersonSummaryInfoResult.Success.Full -> "${offenderSummary.summary.name.forename} ${offenderSummary.summary.name.surname}"
+          is PersonSummaryInfoResult.Success.Restricted -> "LAO Offender"
+          else -> "Unknown"
```

Comment:
we _could_ use the new OffenderDetailsUtil#getNameFromPersonSummaryInfoResult here which would be cleaner, but It would slightly change the behaviour:

- we'd include middle names in the result (presumably this is a good thing)
- for not found we'd return "Not Found" (instead of 'Unknown')

Any thoughts from a reviewer?

Reply:
I think we should probably update `getNameFromPersonSummaryInfoResult` to return `Unknown`, just for consistency. I agree on middle names too 👍

Reply:
Done